### ISSUE TITLE
Get latest UTA transcripts and be able to use their gap alignments

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#515]
* We use UTA transcripts from 2018
* We didn't retrieve the CIGAR strings, so we set the 1,140 transcripts with gaps to error/not be usable for local HGVS (only 1 prod classification, though)

It might be useful to retrieve their transcripts (including 3 more years worth of transcripts) with CIGAR strings, and thus be able to use them.

I downloaded 140,318 transcripts from UTA via:

```
\copy (SELECT transcript.ac, string_agg(distinct transcript.hgnc, ',') as hgnc, string_agg(distinct origin.url, ',') as origin_url,
string_agg(distinct es.alt_ac::varchar, ',') as contig,
string_agg(distinct es.alt_strand::varchar, ',') as strand,
transcript.cds_start_i,
transcript.cds_end_i,
string_agg(exon.start_i::varchar, ',' order by exon.ord) as exon_starts,
string_agg(exon.end_i::varchar, ',' order by exon.ord) as exon_ends,
string_agg(exon_aln.cigar, ',' order by exon.ord) as cigars
from uta_20210129.transcript transcript
inner join uta_20210129.exon_set es on (transcript.ac = es.tx_ac AND alt_aln_method = 'splign')
inner join uta_20210129.origin origin on (transcript.origin_id = origin.origin_id)
Inner join uta_20210129.exon as exon on (es.exon_set_id = exon.exon_set_id)
inner join uta_20210129.exon_aln exon_aln on (exon_aln.alt_exon_id = exon.exon_id)
WHERE es.alt_ac in
('NC_000001.10', 'NC_000002.11', 'NC_000003.11', 'NC_000004.11', 'NC_000005.9', 'NC_000006.11', 'NC_000007.13', 'NC_000008.10', 'NC_000009.11', 'NC_000010.10', 'NC_000011.9', 'NC_000012.11', 'NC_000013.10', 'NC_000014.8', 'NC_000015.9', 'NC_000016.9', 'NC_000017.10', 'NC_000018.9', 'NC_000019.9', 'NC_000020.10', 'NC_000021.8', 'NC_000022.10', 'NC_000023.10', 'NC_000024.9') and origin.origin_id not in (10, 11)
group by transcript.ac) TO 'uta.csv' CSV HEADER;
```

Investigating their data, their cigar strings are not just different than our (ie providing a D when we use an I) but they appear to have aligned slightly differetly too (perhaps slightly different splign params, or normalization?) - they are in repetitive regions - and they seem to add up, not sure if it matters.

UTA exons etc are done in stranded order, while ours are in genomic order

NM_033493.1 (-) last 1st exon:

1590374, 1590473, 31=1I50=2X15=
1590374, 1590473, M32 D1 M66

Locus of diff: 1:1590442-1590443 (UCSC) del is in repetitive region GCGCGCGCGC

There appears to be a 1D at the end of the last exon as well (polyA tail?)

----------

NM_000804.2 (+) 3rd exon:

71850006, 71850193, 150=2D37=
71850006, 71850193, M151 I2 M36

Locus of diff: 11:71850157-71850158 (1 based) the del is in a repetitive region: TCTC

---------------

NM_000682.5 (-) only exon

96778622, 96781888, 891=9D2375=
96778622, 96781888, M902 I9 M2364

Locus of diff: 2:96780986 - 96780997 in repetitive region: CCTCCT

```
----------
     ac      | hgnc  |             origin_url              |   contig    | strand | cds_start_i | cds_end_i |                 exon_starts                  |                  exon_ends                   |            cigars            
-------------+-------+-------------------------------------+-------------+--------+-------------+-----------+----------------------------------------------+----------------------------------------------+------------------------------
NM_033493.1 | CDK11B | http://www.ncbi.nlm.nih.gov/refseq/ | NC_000001.10 | -1     |         111 |      2427 | 1590374,1588824,1586818 | 1590473,1588948,1586938 | 31=1I50=2X15=,124=,57=1X58=2X2=1D
 NM_000804.2 | FOLR3 | http://www.ncbi.nlm.nih.gov/refseq/ | NC_000011.9 | 1      |          50 |       788 | 71846770,71846992,71850006,71850390,71850630 | 71846814,71847166,71850193,71850526,71850934 | 44=,174=,150=2D37=,136=,304=
 NM_000682.5 | ADRA2B | http://www.ncbi.nlm.nih.gov/refseq/ | NC_000002.11 | -1     |           0 |      1353 | 96778622    | 96781888  | 891=9D2375=

```

**Comment by davmlaw** [comment]
This will get fixed once we've done #566 - cdot for transcript annotations

**Comment by EmmaTudini** [comment]
Waiting on comment on #566

**Comment by davmlaw** [comment]
Not sure what comment was needed in other issue.

But since then, we've introduced the ability to test matching without re-running things, see #651

**Comment by EmmaTudini** [comment]
I don't think we ever re-matched the variants that needed it for some reason? @TheMadBug Can you remember what the blocker was?

**Comment by EmmaTudini** [comment]
I think I remember what it was (and it is a little off topic I think)- re: the "37 resolves as intergenic when it should have been an intron" - this is because Cdot didn't properly convert data from UTA, this only affects resolution when:
UTA transcript is used (way less than 1% of our transcripts)
Variant is off the transcript
Which is why we only found it with that single variant

So we never rematched anything due to waiting on the above fix. Could probably do all of them minus that variant?

**Comment by TheMadBug** [comment]
I don't remember sorry. I have a vague recollection that the main reason we didn't rematch anything in prod was because the data that would have been changed was a bit off in the first place.
I'm running the rematch dry run tool against prod now to see what's outstanding.

**Comment by EmmaTudini** [comment]
@TheMadBug See above - not sure whether another issue was raised?

**Comment by davmlaw** [comment]
[#682 HGVS resolved as off transcript (but is really an intron?)](https://github.com/SACGF/variantgrid/issues/682#issuecomment-1274057889) ???

**Comment by TheMadBug** [comment]
Was the other thing that we wanted a way you could pass in a list of IDs to get rematched?
We have that now, so if we wanted to pick a subset of this file to rematch, we could

[rematching.csv](https://github.com/SACGF/variantgrid/files/9751106/rematching.csv)

**Comment by davmlaw** [comment]
/classification/classification/7229 is the NM_001101.4(ACTB):c.-6-4G>T issue referenced in #682 

Shariant prod was released 7th of September and #682 was fixed 19th september - so is in Shariant test but not prod

**Comment by EmmaTudini** [comment]
Ok perfect so next deploy we can rematch. @TheMadBug Would you run the rematching code over test to see whether any variants change due to the change in #682

**Comment by TheMadBug** [comment]
Here's the updated matchings, 
If you ignore the Ensembl transcript complaints, there's two records that now match and one record that used to that now wont.

[rematching.csv](https://github.com/SACGF/variantgrid/files/9751325/rematching.csv)

**Comment by davmlaw** [comment]
Deployed a while now. Will raise an individual issues if they occur
